packages/wallet/test/services/AssetsTracker.test.ts

```diff
@@ -66,6 +67,59 @@ describe('createAssetsTracker', () => {
     });
   });
 
+  it('does not emit if the assets in total balance have not changed', () => {
+    createTestScheduler().run(({ cold, expectObservable, flush }) => {
+      const balanceTracker = {
+        utxo: {
+          total$: cold('a-b-c', {
+            a: {} as Cardano.Value,
+            b: { assets: new Map([[AssetId.TSLA, 1n]]), coins: 1_000_000n } as Cardano.Value,
+            c: { assets: new Map([[AssetId.TSLA, 1n]]), coins: 2_000_000n } as Cardano.Value
+          })
+        }
+      } as unknown as TransactionalTracker<BalanceTracker>;
+
+      const target$ = createAssetsTracker(
+        { assetProvider, balanceTracker, logger, retryBackoffConfig } as unknown as AssetsTrackerProps,
+        {
+          assetService
+        }
+      );
+      expectObservable(target$).toBe('a-b--', {
+        a: new Map(),
+        b: new Map([[AssetId.TSLA, assetTsla]])
+      });
+      flush();
+      expect(assetProvider.setStatInitialized).toBeCalledTimes(1);
+      expect(assetService).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  it('emits at least once if there are no assets found in total balance', () => {
+    createTestScheduler().run(({ cold, expectObservable, flush }) => {
+      const balanceTracker = {
+        utxo: {
+          total$: cold('a-b-c', {
+            a: { coins: 1_000_000n } as Cardano.Value,
+            b: { coins: 2_000_000n } as Cardano.Value,
+            c: { coins: 3_000_000n } as Cardano.Value
+          })
+        }
+      } as unknown as TransactionalTracker<BalanceTracker>;
+
+      const target$ = createAssetsTracker(
+        { assetProvider, balanceTracker, logger, retryBackoffConfig } as unknown as AssetsTrackerProps,
+        {
+          assetService
+        }
+      );
+      expectObservable(target$).toBe('a----', { a: new Map() });
+      flush();
+      expect(assetProvider.setStatInitialized).toBeCalledTimes(1);
+      expect(assetService).not.toHaveBeenCalled();
```

Comment:
These assertions are probably redundant since other test covers it. It's fine to keep though.